service/commentutil/commentutil.go

```diff
@@ -77,6 +77,35 @@ func MarkdownComment(c *reviewdog.Comment) string {
 	return sb.String()
 }
 
+// MarkdownSuggestions creates diff in markdown for suggested changes
+func MarkdownSuggestions(c *reviewdog.Comment) string {
+	suggestions := c.Result.Diagnostic.GetSuggestions()
+	if len(suggestions) == 0 {
+		return ""
+	}
+
+	var sb strings.Builder
+
+	suggestionsRendered := 0
+	for _, s := range suggestions {
+		if s.Range == nil || s.Range.Start == nil || s.Range.End == nil {
+			continue
+		}
+		if suggestionsRendered > 0 {
+			sb.WriteString("\n\n")
+		}
+		sb.WriteString("```suggestion:-0+")
```

Comment:
Suggestion position (lines) is not always the same as the position of the diagnostic.
Can you test all possible cases including multiple lines suggestions (including E2E testing)? You can refer to the test of GitHub.

It might be difficult to support all cases, but we shouldn't post wrong suggestions as much as possible.